### ISSUE TITLE
Chapter 8 some suggestions and possible mistakes

### ISSUE THREAD

**Issue by sebastianbertoli** [mml-book/mml-book.github.io#426]
**Is your feature request related to a problem? Please describe.**
In my view chapter 8, while being very useful, feels less polished / consistent with the other chapters. I hope that some of my points are useful. :-)

1. There is no general further reading section, but each subsection has its own further reading section which is not visible in the table of contents.
1. I find the interchanging use of "future data" and "unseen data" could be confusing to the reader since "future data" somehow implies a temporal ordering.
1. Equation 8.6 and following have the R in **R**_emp in boldface. I am not sure if that is intentional.
1. p. 261, Equation 8.10 could also show the explicit formula for the expected risk with the pdf/pmf of the data.
1. p. 265, 8.2.5 Further Reading has the following paragraph which perhaps would be better suited on p.261 close to equation 8.10 (see my previous point).

> Thinking about empirical risk minimization (Section 8.2) as “probabil-
> ity free” is incorrect. There is an underlying unknown probability distri-
> bution p(x,y) that governs the data generation. However, the approach
> of empirical risk minimization is agnostic to that choice of distribution.
> This is in contrast to standard statistical approaches that explicitly re-
> quire the knowledge of p(x,y) . Furthermore, since the distribution is a
> joint distribution on both examples x and labels y , the labels can be non-
> deterministic. In contrast to standard statistics we do not need to specify
> the noise distribution for the labels y .

**Describe the solution you'd like**
1. If possible structure the section like the others(?)
1. Drop the term future data.
1. If it was intentional, maybe add a note in the table of symbols? I was not sure how to interpret the boldface.
1. Expand expectations to explicitly show dependence on the unknown probability distribution.
1. Move paragraph to p.261 where the expected risk is explained.

**Comment by mpd37** [comment]
Thanks for the feedback, which is very helpful. Let me start commenting on some issues; @chengsoonong can continue later.

Generally, chapter 8 is different from all other chapters in the sense that it bridges part 1 and part 2 of the book and touches upon a range of fairly different topics.

1. We decided to have a separate further reading section after each of those sub-topics instead of a multi-page further-reading section at the end. Given that the topics covered in chapter 8 are not directly connected, this makes more sense to us. These subsections "further reading" don't show up in the TOC because subsections don't show up in any other chapters.

**Comment by chengsoonong** [comment]
2. The word "unseen" is unfortunately also ambiguous. It could mean unobserved or hidden variables. A few sentences later, we say "future unseen data", to try to resolve this confusion. Unfortunately, I think this is hard to get perfectly clear with English.

3. I assume you mean that because it is bold, it is somehow a vector? It is a scalar. It does appear in quite a few places, and it would be quite difficult to fix at this point in production.

**Comment by chengsoonong** [comment]
4. and 5. You mean to expand out the definition of the expectation? I am not sure what that really adds to the understanding. Expanding out the definition comes with an additional possible confusion of requiring the reader to recall notation for pmf/pdf.

We debated about whether to bring this idea about the distribution p(x,y) to the front at that point in the text, when this chapter was written. In the end, we felt that a shorter description (chapter is already pretty long), without too many side comments is a better way to go. It is a matter of taste.

**Comment by chengsoonong** [comment]
Thank you so much for your detailed suggestions. It is difficult for us to make major changes in the book since it is in production. Perhaps we can do something about a future edition (if the book is popular enough).

**Comment by sebastianbertoli** [comment]
Hi, thanks for the replies! I agree, most of my points are indeed a matter of taste! Good point about (2), I must admit I did not consider that. (3) Ok, that is what I thought. (5) Well, I was reading this chapter and the Wiki page/Murphy in parallel and personally found it nice to have the expectation expanded to make things super-clear.  (5) I understand what you mean. My motivation for having raised that point is that I often skip further reading sections because I do not expect to find further explanations in them, just pointers to additional literature and topics. 

Overall, none of these points really detract from the book. I thought I make some suggestions in case they are useful. It also helps me while reading the book. :)

On a sidenote, the book is amazing and I find it very cool that you guys put the draft on GitHub for feedback. I wish all ML books were made like this.  I am sure the book has the potential to become very popular and I am definitely recommending it to all my friends interested in the topic.

P.S. I might open further issues while I read through the remaining chapters but will focus on actual mistakes which I assume can be fixed easily (typos and such).
